workitem/workitem_repository.go

```diff
@@ -119,27 +121,77 @@ func (r *GormWorkItemRepository) Delete(ctx context.Context, ID string) error {
 	return nil
 }
 
-// Reorder places the to-be-reordered workitem(s) above the input workitem.
+// Calculates the order of the to-be-reordered workitem
+func (r *GormWorkItemRepository) CalculateOrder(above, below *float64) float64 {
+	itemOrder := (*above + *below) / 2
+	return itemOrder
+}
+
+// FindSecondItem returns the order of the second workitem required to reorder.
+// If position.direction == "above", then
+//	FindItemItem returns the value above which reorder item has to be placed
+//      FindSecondItem returns the value below which reorder item has to be placed
+// If position.direction == "below", then
+//	FindItemItem returns the value below which reorder item has to be placed
+//      FindSecondItem returns the value above which reorder item has to be placed
+func (r *GormWorkItemRepository) FindSecondItem(order *float64, secondItem string, WorkItemId string) (*uint64, *float64, error) {
```

Comment:
@kwk that's because I just need the order of secondItem. Also, I can reuse the tx errors; otherwise I will have to write them twice in Reorder() function